plugins/include/nvault.inc

```diff
@@ -21,43 +21,128 @@
 	#pragma loadlib nvault
 #endif
 
-/* All timestamps are in UNIX epoch form. */
+/**
+ * All timestamps are in UNIX epoch form.
+ */
 
-/* Opens a vault by name (such as "myvault")
- * Returns a vault id, INVALID_HANDLE otherwise (-1)
+/**
+ * Opens a vault by name. Creates a vault if it doesn't exist yet.
+ *
+ * @param name			Name of the vault. The vault will be created in
+ * 						/amxmodx/data/vault directory.
+ *
+ * @return              The vault handle to be used in other natives.
+ * 						INVALID_HANDLE (-1) if not successfully opened.
  */
 native nvault_open(const name[]);
 
-/* Gets a vault value by returning an int
- * setting a byref float or setting a string + maxlength
+/**
+ * Retrieves a value from the given key
+ * 
+ * @note An example of retrieving a string:
+ * 		 nvault_get(vaultHandle, "myKey", myString, charsmax(myString));
+ *
+ * @param vault			A vault handle returned from nvault_open()
+ * @param key			A key to get the value from
+ * @param ...			If three argument are given, gets a float value and
```

Comment:
These type of natives are awkward to document, but I think this is a cleaner way to do it: https://github.com/alliedmodders/amxmodx/blob/master/plugins/include/amxmodx.inc#L734-L741